web/server/codechecker_server/server.py

```diff
@@ -912,6 +915,46 @@ def add_product(self, orm_product, init_db=False):
 
         self.__products[prod.endpoint] = prod
 
+    def get_if_database_in_use(self, product):
+        """
+        Returns the product endpoint if the given database is in use
+        """
+        # get the database name from the product connection string
+        is_sqlite = False
+        if product.database.endswith('.sqlite'):
+            to_add = product.database.replace('////', '/')
+            is_sqlite = True
+        else:
+            to_add = f"@{product.host}:{product.port}/{product.database}"
+
+        LOG.info("The database string that will be added: %s", to_add)
+
+        LOG.info("to add: %s", to_add)
+
+        # get the current state of connected databases from products
+        dynamic_list = [
+            a.connection.replace('////', '/')
+            for a in self.cfg_sess_private.query(ORMProduct).all()
+        ]
+        self.cfg_sess_private.commit()
+        self.cfg_sess_private.close()
+
+        # remove the first 16 charecters because in query it is
+        # included in the path string
+        dynamic_list = [
+            d[16:] if d.endswith('.sqlite') else d
```

Comment:
We shouldn't rely on the length of a specific connection string prefix. `d.split(':', 1)` would split the connection string on the first `:` character.

Reply:
fixed